src/block-support/supported-blocks.json

```diff
@@ -0,0 +1,36 @@
+{
+	"common": [
+		"core/paragraph",
+		"core/heading",
+		"core/more",
+		"core/image",
+		"core/video",
+		"core/nextpage",
+		"core/separator",
+		"core/list",
+		"core/quote",
+		"core/media-text",
+		"core/preformatted",
+		"core/gallery",
+		"core/columns",
+		"core/column",
+		"core/group",
+		"core/freeform",
+		"core/button",
+		"core/spacer",
+		"core/shortcode",
+		"core/buttons",
+		"core/latest-posts",
+		"core/verse",
+		"core/cover",
+		"core/social-link",
+		"core/social-links",
```

Comment:
Instead of having to duplicate all the "core" blocks both here and in gutenberg's `packages/block-library/src/index.native.js` file, could we just have the gutenberg JS expose the list of "core" blocks to the native code via the JS bridge? Then we would only need to worry about non-core blocks in this file and avoid having to keep the list of supported blocks in sync. 

Having said that, I realize that this is a temporary solution that we intend to refactor later, so I'm open to leaving this as-is.

Reply:
I agree I would very much like to achieve that goal however I'm a bit stuck on how we might be able to do it. The list here contains the block name but the [registration](https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/index.native.js#L194) ends up taking the block its self and [using](https://github.com/WordPress/gutenberg/blob/5dc434cd951c51d7965179fb84719ebb0ee219b8/packages/block-library/src/index.native.js#L129) the `metadata`, `settings`, and `name`. This is probably just a result of me still catching up with JS but I'm currently not sure what the best way would be to convert the block names to what we need. 

We would need to do something like what's below. I'm just not entirely sure how to handle the pseudo-code part

```
const registerBlock = ( blockName ) => {
	if ( ! blockName ) {
		return;
	}

        //  pseudo-code 
        const block =  import * from `./${blockName.split("/").slice(-1)}`
        // end pseudo-code 

	const { metadata, settings, name } = block;
	registerBlockType( name, {
		...metadata,
		...settings,
	} );
};
```

Reply:
> I'm currently not sure what the best way would be to convert the block names to what we need.

Instead of converting the names into blocks, could we have gutenberg convert the blocks into names for us [along these lines](https://github.com/WordPress/gutenberg/commit/9da350162f8c9f6fea602cee10f34dd36c8ea1fc) and give the native side access to those names?

Reply:
That would be a good way to go as well. The only hardship here is we need this before the editor starts (or maybe the editor never even startes). So ideally we would have a script that could start up a headless editor export the results of that `console.log` to a file then we'd be able to consolidate in some way that way. I hit a bit of a dead end on that though 😞

Reply:
> The only hardship here is we need this before the editor starts (or maybe the editor never even starts).

Ah yes, I didn't think about that. Good point!

Reply:
I'm just brainstorming at this point, but would it work to just hardcode the block names to a variable in the same file in gutenberg? That way we would just be keeping two variables in the same file in sync instead of keeping two files across two repos in sync?

To be clear, I do **_not_** consider this a blocking issue for this PR. Just talking through different ideas.

Reply:
>To be clear, I do not consider this a blocking issue for this PR. Just talking through different ideas.

No problem at all I appreciate the extra thoughts!

>just hardcode the block names to a variable in the same file in gutenberg?

This could be viable. We would want Gutenberg and Gutenberg-mobile to have a list of supported blocks. `Gutenberg` would serve up the list that handles all of the `core` blocks and `Gutenberg-mobile` would handle the subset of `jetpack` blocks that it supports. With that pattern we might want to say something along the lines of "plugins should provide a list of blocks supported on mobile" and we consolodate those in `Gutenberg-mobile`

Reply:
> So ideally we would have a script that could start up a headless editor export the results of that console.log to a file then we'd be able to consolidate in some way that way. I hit a bit of a dead end on that though 😞

I think there is a way to do this with Webpack but I'm not 100% sure yet